### ISSUE TITLE
s0.loss_rpn_reg is 0

### ISSUE THREAD

**Issue by ZhengyuZhang96** [thangvubk/Cascade-RPN#6]
hi,Author, thank you for your wonderful code！but when i am training cascade_rpn, i found that **s0.loss_rpn_reg**  was always 0 . Is that correct?
![sdadadad](https://user-images.githubusercontent.com/49669922/70497782-460f5e00-1b4f-11ea-9c58-5ec4b0e9f698.JPG)

**Comment by thangvubk** [comment]
The ``s0.loss_rpn_reg`` should not be 0. Did you modify the training config.

**Comment by studyharderer** [comment]
Thank you for your reply! I just used a new config file downloaded from your code to train again. and i just changed the location of training set. The result is the same:**s0.loss_rpn_reg** is always 0.  How can i solve it ?

**Comment by ZhengyuZhang96** [comment]
Thank you for your reply! I just used a new config file downloaded from your code to train again. and i just changed the location of training set. The result is the same:s0.loss_rpn_reg is always 0. How can i solve it ? @thangvubk

**Comment by thangvubk** [comment]
I didnt face this training problem. Did you train the network with 8 GPUs?
Here is the training log.
[CRPN.log](https://github.com/thangvubk/Cascade-RPN/files/3963512/CRPN.log)

**Comment by thangvubk** [comment]
Base on the number of batches in 1 epoch in your comment, I guess you are not using 8GPUs for training. Please check [Learning Rate Schedule](https://github.com/thangvubk/Cascade-RPN/blob/master/docs/GETTING_STARTED.md#train-a-model) for the solution.

**Comment by ZhengyuZhang96** [comment]
oh yes, i am using single gpu for training,but i had already followed **Learning Rate Schedule**. Specifically, i changed the lr to 0.0025 and keep img_per_gpu equal to 2. I wonder if only multi-GPU training can make s0.loss_rpn_reg non-zero ? Can your try training by single gpu in your environment ? Looking forward to your reply ! @thangvubk

**Comment by thangvubk** [comment]
I can train with single GPU normally with lr=0.0025.
![image](https://user-images.githubusercontent.com/20570053/70972108-f1875800-20e5-11ea-8e4d-4eec4ce2175e.png)

**Comment by ZhengyuZhang96** [comment]
Thank you very much for your reply. I rebuilt the environment and used the same config file. Now everything is normal!

**Comment by jiyoon0630** [comment]
> Thank you very much for your reply. I rebuilt the environment and used the same config file. Now everything is normal!

Hi @studyharderer Can you explain more about the solution "I rebuilt the environment and used the same config file."? Cause, I've changed lr to 0.0025 and re-clone this git, however, loss is still 0.

**Comment by sunpeng1996** [comment]
+1

**Comment by sunpeng1996** [comment]
Hi, Thank you for your great work!
  In my experiments, s0.loss_rpn_reg is always 0. 
  I reattached the package according to the issues on Github, but it still doesn't work. Could you please tell me which library caused it?
Thanks

**Comment by jiyoon0630** [comment]
@sunpeng1996
Hi. Check #8 .
I still don't know which library causes the problem, however, in my case, setting up virtualenv and reinstall everything solved it!

**Comment by sunpeng1996** [comment]
Hi, Thank you for your reply. @jiyoon0630 @thangvubk 

In fact, I have tried according to that issue, but the problem has not been solved. I aligned the package versions of the environment one by one, but the problem remained.
